rpcs3/Emu/Memory/vm.cpp

```diff
@@ -270,16 +270,26 @@ namespace vm
 		return result;
 	}
 
-	static void _lock_shareable_cache(u64 flags, u32 addr, u32 size)
+	static void _lock_main_range_lock(u64 flags, u32 addr, u32 size)
 	{
-		// Can't do 512 MiB or more at once
-		if (size >= 1024 * 1024 * 512)
+		// Shouldn't really happen
+		if (size == 0)
+		{
+			vm_log.warning("Tried to lock empty range (flags=0x%x, addr=0x%x)" HERE, flags >> 32, addr);
+			g_range_lock.release(0);
+			return;
+		}
+
+		// Limit to 256 MiB at once; make sure if it operates on big amount of data, it's page-aligned
+		if (size > 256 * 1024 * 1024 || (size > 65536 && size % 4096))
 		{
 			fmt::throw_exception("Failed to lock range (flags=0x%x, addr=0x%x, size=0x%x)" HERE, flags >> 32, addr, size);
 		}
 
 		// Block or signal new range locks
 		g_range_lock = addr | u64{size} << 35 | flags;
+
```

Comment:
?

Reply:
Currently irrelevant, but will be needed later.